effects/butterfly-gears/butterfly-gears.c

```diff
@@ -0,0 +1,167 @@
+#include "startup.h"
+
+#include "hardware.h"
+#include "interrupts.h"
+#include "blitter.h"
+#include "color.h"
+#include "copper.h"
+#include "bitmap.h"
+#include "palette.h"
+#include "pixmap.h"
+#include "memory.h"
+#include "sprite.h"
+#include "fx.h"
+
+#define ROTZOOM_W 24
+#define ROTZOOM_H 24
+#define COPWAIT_X 1
+#define Y0 Y((256-280)/2)
+#define COPPER_HALFROW_INSTRUCTIONS ROTZOOM_W/2+2
+#define INSTRUCTIONS_PER_BALL COPPER_HALFROW_INSTRUCTIONS*2*ROTZOOM_H
+#define DEBUG_COLOR_WRITES 0
+#define USE_DEBUG_BITMAP 0
+
+#if DEBUG_COLOR_WRITES // only set background color for debugging
+#define COLREG_MASK 0
+#else
+#define COLREG_MASK 0xff
+#endif
+
+typedef struct {
+  PixmapT texture;
+  short angle;
+  short angleDelta;
+  int uvPos; // TODO handle u,v separately
+  int uvDelta;
+} BallT;
+
+typedef struct {
+  CopListT *cp;
+  CopInsT *upperBallCopper;
+  CopInsT *lowerBallCopper;
+} BallCopListT;
+
+static BallCopListT ballCopList1; // TODO use second copper list and double-buffer
+static BallT ball1;
+static BallT ball2;
+
+#if USE_DEBUG_BITMAP
+#include "data/gears_testscreen_debug.c"
+#else
+#include "data/gears_testscreen.c"
+#endif
+#include "data/texture_butterfly.c"
+#include "data/texture_butterfly2.c"
+
+// Create copper writes to color registers, leave out colors needed for sprites
+static void InitCopperListBall(CopListT *cp, int y) {
+  short i;
+
+  for (i=0; i<ROTZOOM_H; i++) {
+    CopWait(cp, y, COPWAIT_X);
+    CopMove16(cp, color[ 3 & COLREG_MASK], 0xf00);
```

Comment:
Don't hesitate to use macros in such cases to save on boilerplate code. I would just define and use following:
```
#define SETCOLOR(x) CopMove16(cp, color[(x) & COLREG_MASK], 0xf00)
```
Oh... if not, then at least I'd consider a good idea to introduce:
```
#if DEBUG_COLOR_WRITES // only set background color for debugging
#define COLREG(x) 0
#else
#define COLREG(x) (x)
#endif
```
... to replace that confusing (at first glance) trick with `COLREG_MASK`.

Reply:
Macros added 👍